### ISSUE TITLE
[Proposal] Default outputs via `output_of` actor class method

### ISSUE THREAD

**Issue by llhhaa** [sunny/actor#151]
In a project I'm working on, we've taken to using actors as a way to isolate and encapsulate complex bits of fetching logic. For example:

```ruby
class GetVersionChanges < ApplicationActor
  input :version
  input :filters, default: {}
  output :output

  def call
    # complicated database calls
    # version_changes = ...
    this.output = version_changes
  end
end

GetVersionChanges.call(version:, filters:).output
```

Calling `.call().output` for these gets a bit tiring as use of the pattern grows throughout the app. What I'd like to do is something like this:
```ruby
class GetVersionChanges < ApplicationActor
  input :version
  input :filters, default: {}

  def call
    # complicated database calls
    version_changes # the return value is picked up and returned by `output_of`
  end
end

GetVersionChanges.output_of(version:, filters:)
```

In fact, I've monkey-patched the gem with the following to get this behavior in our application:
```ruby
module ServiceActor::Core
  module ClassMethods
    # Get the return value of calling an actor with arguments.
    #
    #   CreateUser.output_of(name: "Joe")
    def output_of(result = nil, **arguments)
      call(result, **arguments)[:_default_output]
    end
  end

  def _call
    result._default_output = call
  end
end
```

It's rudimentary - "polluting" the result object with a `_default_output` attribute - but it works. Let me know if 1) this feature is a good fit for the gem and 2) if this implementation is sufficient, or if something more sophisticated is desired. Thanks!

**Comment by sunny** [comment]
Hi there! Thank you for bringing up this suggestion 🤗

While I understand the need for a more streamlined syntax I don’t think this is a good fit for the gem as it doesn’t work well with the idea of a single type of signature for actors, doesn’t allow for actors to be used in `play` and pass their outputs from one to the other, doesn’t allow for the `fail!` and `rollback` mechanisms.

While I think this is fine as a monkey-patch I’d rather not support it fully by adding this in the gem. As an extra precaution if you are using this I’d suggest adding `output :_default_output` to your `ApplicationActor`.

**Comment by llhhaa** [comment]
Thanks for the response!

I appreciate your concerns, the initial implementation was indeed rudimentary. I was already working on the more sophisticated implementation in #152, and I think it addresses almost all of your concerns, so went ahead and opened it.

- Supports `play` by returning the return value of the final actor in the chain.
- Supports `fail!`, as it raises normally when an actor or play chain actor fails.
- Supports `rollback`, though like `call` it will raise an error, and to access the failed actor state one still needs to use `result`.
  - I have a notion to add something like a `last_rollback_value` option, though I think this use case is niche enough that using `result` suffices.

By single type signature, I understand you to mean that `call` and `result` both return a `ServiceActor::Result`. I'll suggest that `output_of` is purely additive, it still leverages results, doesn't impinge on existing functionality, and just streamlines the accessing of `result.explicit_output`.

**Comment by sunny** [comment]
Closed by [#152](https://github.com/sunny/actor/pull/152). This is now possible using the `.value(…)` method 👏🏻
